lib/class/art.class.php

```diff
@@ -1103,8 +1103,11 @@ public function gather($options = array(), $limit = 0)
 
             return array();
         }
-        $config  = AmpConfig::get('art_order');
-        $methods = get_class_methods('Art');
+        if ($limit == 0) {
+            $limit   = (is_null(AmpConfig::get('art_search_limit'))) ? 5 : AmpConfig::get('art_search_limit') ;
```

Comment:
Wouldn't it be a idea to define a public class constant for the default value (5) instead of having this `magic int` in several places? Any later change can simply be done by changing the constant.

Reply:
I may be wrong, but I thought that access to the various gather functions is directed through ` gather($options = array(), $limit = 0)` thus the  "magic" number is not actually needed for the default parameter value in the other functions.  If not, perhaps all calls **should** be directed through this "gateway".

But then the functions called on by gather  would need to be private to prevent bypassing by external functions.  Would the constant then be advantageous?

Reply:
You're right with your assumption, that it `should` be that way. If someone re-uses the code in future, this may change (maybe by some classic copy-paste).

As long as those numbers represent the default value, a constant is imo very advantageous. Tomorrow, some one else needs to define this art-search limit in another place - that would be the third occurrence (after lines 1107 and 1196).

If one then decides to change the default value, one has to _know_ these 3 occurrences and has to change them by hand. If one has a constant, one can change it with a single line of code.

Just an idea:
In an ideal world, every default value from the config would have a constant and all those constants would be placed in the same place (an `Enumeration`, but php does not support real enum datatypes :( ).

It would look as follows:
```
final class ConfigDefaults {
  public const ART_SEARCH_LIMIT = 5;
  public const SPOTIFY_ART_FILTER_LIST = "artist";
...
}
```
Some time, the default config file could be generated using these defaults, too.

Reply:
There are more $limit=5 in the other class functions than those two.

I did a grep andl found only `gather($options = array(), $limit = 0)` called externally.  The defaults for all the other functions called by this function should not need a default value for $limit.

But I like the idea of descriptive class constants.  Perhaps this could be added to the wiki coding standards page.